### ISSUE TITLE
Bug in HighPassFilter

### ISSUE THREAD

**Issue by codeflakes0** [nickgillian/grt#99]
Just realized why I kept getting zeros.
In HighPassFilter.cpp

`VectorFloat HighPassFilter::filter(const VectorFloat &x){ 
...
        yy[n] = filterFactor * (yy[n] + x[n] - xx[n]) * gain;      
        xx[n] = x[n];
        processedData[n] = processedData[n];
...
}`

should be something like this.

`yy[n] = filterFactor * (yy[n] + x[n] - xx[n]) * gain;      
xx[n] = x[n];
processedData[n] = yy[n];`

I did not have time to test it yet but I'm sure there's a problem there.

**Comment by nickgillian** [comment]
Yes, this was a bug in the high pass filter class, thanks for catching this!

I've pushed a fix to the dev branch, so if you check that out you can get direct access to the patch.  I'll merge this to the master branch in a few days when some other changes in the dev branch have been cleaned up.
